### PR TITLE
Implement 1 UNG card

### PR DESCRIPTION
**Pull request by packkill**
This revision includes:

- Implement 1 UNG card

  - 'Devilsaur Egg' (UNG_083)

### CODE REVIEW

**Comment by packkill** [comment]
I manually changed the style in the last PR.

File: Sources/Rosetta/PlayMode/CardSets/UngoroCardsGen.cpp [hunk 1]
```diff
@@ -1850,6 +1850,10 @@ void UngoroCardsGen::AddNeutral(std::map<std::string, CardDef>& cards)
     // GameTag:
     // - DEATHRATTLE = 1
     // --------------------------------------------------------
+    cardDef.ClearData();
+    cardDef.power.AddDeathrattleTask(
+        std::make_shared<SummonTask>("UNG_083t1", SummonSide::DEFAULT));
```
Comment (utilForever) [inline Sources/Rosetta/PlayMode/CardSets/UngoroCardsGen.cpp:1855]:
```suggestion
        std::make_shared<SummonTask>("UNG_083t1", SummonSide::DEATHRATTLE));
```

File: Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp [hunk 2]
```diff
@@ -311,6 +311,104 @@ TEST_CASE("[Warlock : Minion] - UNG_833 : Lakkari Felhound")
...
+    opPlayer->SetTotalMana(10);
+    opPlayer->SetUsedMana(0);
+
+    auto& curField = *(curPlayer->GetFieldZone());
+    auto& opField = *(opPlayer->GetFieldZone());
+
+    const auto card1 =
+        Generic::DrawCard(curPlayer, Cards::FindCardByName("Devilsaur Egg"));
+    const auto card2 = Generic::DrawCard(
+        opPlayer, Cards::FindCardByName("Wolfrider", FormatType::CLASSIC));
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp:349]:
```suggestion
    const auto card2 = 
        Generic::DrawCard(opPlayer, Cards::FindCardByName("Wolfrider"));
```

File: Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp [hunk 3]
```diff
@@ -311,6 +311,104 @@ TEST_CASE("[Warlock : Minion] - UNG_833 : Lakkari Felhound")
...
+    game.Process(opPlayer, AttackTask(card2, card1));
+
+    CHECK_EQ(curField.GetCount(), 1);
+    CHECK_EQ(curField[0]->card->name, "Devilsaur");
+    CHECK_EQ(curField[0]->GetAttack(), 5);
+    CHECK_EQ(curField[0]->GetHealth(), 5);
+
+    CHECK_EQ(opField[0]->card->name, "Wolfrider");
+    CHECK_EQ(opField[0]->GetAttack(), 3);
+    CHECK_EQ(opField[0]->GetHealth(), 1);
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp:369]:
```suggestion
    game.Process(opPlayer, PlayCardTask::Minion(card2));
    game.Process(opPlayer, AttackTask(card2, card1));
    CHECK_EQ(curField.GetCount(), 1);
    CHECK_EQ(curField[0]->card->name, "Devilsaur");
    CHECK_EQ(curField[0]->GetAttack(), 5);
    CHECK_EQ(curField[0]->GetHealth(), 5);
```

File: Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp [hunk 4]
```diff
@@ -311,6 +311,104 @@ TEST_CASE("[Warlock : Minion] - UNG_833 : Lakkari Felhound")
...
+
+    auto& curHand = *(curPlayer->GetHandZone());
+
+    const auto card1 =
+        Generic::DrawCard(curPlayer, Cards::FindCardByName("Fire Fly"));
+
+    game.Process(curPlayer, PlayCardTask::Minion(card1));
+
+    CHECK_EQ(curHand.GetCount(), 5);
+    CHECK_EQ(curHand[4]->card->name, "Flame Elemental");
```
Comment (utilForever) [inline Tests/UnitTests/PlayMode/CardSets/UngoroCardsGenTests.cpp:409]:
```suggestion
    game.Process(curPlayer, PlayCardTask::Minion(card1));
    CHECK_EQ(curHand.GetCount(), 5);
    CHECK_EQ(curHand[4]->card->name, "Flame Elemental");
```

**Review by utilForever — CHANGES_REQUESTED** [review]
You should fix the summon side. 🦖

**Comment by packkill** [comment]
수정했습니다.

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for your work! 🎉
I'll merge it after CIs are passed.

**Pull request merged**